scipy/optimize/_zeros_py.py

```diff
@@ -1392,3 +1392,125 @@ def toms748(f, a, b, args=(), k=1,
                           maxiter=maxiter, disp=disp)
     x, function_calls, iterations, flag = result
     return _results_select(full_output, (x, function_calls, iterations, flag))
+
+
+def _bracketer(f, a, b=None, min=None, max=None, factor=2, batch=100):
+    """Bracket the root of a monotonic scalar function of one variable
+
+    Parameters
+    ----------
+    f : callable
+        The function for which the root is to be bracketed.
+    a, b : float
+        Starting guess of bracket, which need not contain a root. If `b` is
+        not provided, ``b = a + 1``.
+    min, max : float
+        Minimum and maximum allowable endpoints of the bracket, inclusive.
+    factor : float
+        The factor used to grow the bracket. See notes for details.
+    batch : int
+        The maximum number of arguments passed to the callable `f` at once.
+
+    Returns
+    -------
+    l, r : float
+        The left and right endpoints of the bracket such that
+        ``f(l) < 0 < f(r)``.
+
+    Notes
+    -----
+    This function generalizes an algorithm found in pieces throughout
+    `scipy.stats`. The strategy is to iteratively grow the bracket `(l, r)`
+     until ``f(l) < 0 < f(r)``.
+
+    - If `min` is not provided, the distance between `b` and `l` is iteratively
+      increased by `factor`.
+    - If `min` is provided, the distance between `min` and `l` is iteratively
+      decreased by `factor`. Note that this *increases* the bracket size.
+
+    Growth of the bracket to the right is analogous.
+
+    Growth of the bracket in one direction stops when the endpoint is no longer
+    finite, the function value at the endpoint is no longer finite, or the
+    endpoint reaches its limiting value (`min` or `max`). Iteration terminates
+    when the bracket stops growing in both directions, the bracket surrounds
+    the root, or a root is found (accidentally).
```

Comment:
Not sure. Would be good to consider getting this information out somehow.

Reply:
You do get this information. 
* If a bracket is found, you get the lower and upper ends of the bracket.
* If the root is found accidentally, the lower and upper ends of the bracket are equal to the root.
* If no valid bracket is found, you get `(None, None)`.